ui/src/app/shared/services/workflows-service.ts

```diff
@@ -241,22 +241,43 @@ export const WorkflowsService = {
                     }
                     throw new Error('no artifact logs are available');
                 }
-
                 return from(requests.get(this.getArtifactLogsPath(workflow, nodeId, container, archived)));
             }),
             mergeMap(r => r.text.split('\n')),
+            filter(x => !!x), // check for empty content
             map(content => ({content, podName: workflow.status.nodes[nodeId].displayName}) as LogEntry),
             filter(x => !!x.content.match(grep))
         );
     },
 
+    getContainerLogsFromArtifacts(workflow: Workflow, container: string, grep: string, archived: boolean): Observable<LogEntry> {
+        // Iterate over all nodes and check if node is a pod. Then sort nodes by startedAt.
+        const podNodeIds: string[] = Object.keys(workflow.status.nodes)
+            .filter(nodeId => {
+                return workflow.status.nodes[nodeId].type === 'Pod';
+            })
+            .sort((a, b) => {
+                return new Date(workflow.status.nodes[a].startedAt).getTime() - new Date(workflow.status.nodes[b].startedAt).getTime();
+            });
+
+        // Iterate over pod node ids and get their logs
+        return from(podNodeIds).pipe(
+            concatMap(nodeId => {
+                return this.getContainerLogsFromArtifact(workflow, container, nodeId, container, grep, archived);
+            })
+        );
+    },
```

Comment:
I actually think this is potentially problematic -- this could result in tens, hundreds, or thousands of network requests depending on the Workflow. We shouldn't be knowingly creating a thundering herd.

This doesn't quite seem like a fix either, this seems like a feature to enable "All" logs for archive logs (which gets around the "Should specify a node" error above that was added in #11420). The node requirement means there is only one network request at a time.

While getting "All" logs for archive logs seems like a good feature, this implementation is perhaps too naive for large workflows.
Also note that we recommend against using archive logs per #12597 and so I'm a bit hesitant to add more features to a not recommended feature, especially ones that can cause a thundering herd.